### ISSUE TITLE
[bug] A couple of UX improvements

### ISSUE THREAD

**Issue by ghost** [conan-io/conan#13295]
### Environment details

* Operating System+version: any
* Compiler+version: any
* Conan version: 2.0.0
* Python version: any

### Steps to reproduce

1. `conan install` now apparently takes *only* the path to the dir containing conanfile. However, `conan install` without any arguments gives the following error message
```
ERROR: Please specify at least a path to a conanfile or a valid reference.
```
I was told that the reference still can be added via `conan install --requires=<reference>`, but that wasn't by any means obvious. I suggest having something like this
```
ERROR: Please specify a path to a conanfile. To install a reference, use --requires=<reference>.
```

2.  When dealing with a recipe that is not 2.0-ready, there’s an error line
```
If it is your recipe, check it is updated to 2.0
```
I am not a native English speaker, but it sounds to me that it should be “check *if* it is updated” or “check *that* it is updated”

3. ~I have a perfectly valid `conanfile.txt` from 1.X~ already fixed
```
❯ cat conanfile.txt
[requires]
sdl/[^2.26.1]

[options]
sdl:x11=False
```
On 2.0, `conan install` fails with
```
ERROR: Error while parsing [options] in conanfile
Options should be specified as 'pkg:option=value'
```
It took me a couple of minutes to understand that in Conan 2.0 the syntax should be `sdl/*:x11=False`.
While I totally understand that for Conan devs this message looks totally logical, I wish it was clearer for people who migrated from 1.X

**Comment by ghost** [comment]
I could've created 3 separate issues so that PRs could have addressed those easier. However, I didn't want to spam the issue tracker with small cosmetic issues. If you still think creating 3 separate issues is preferable, let me know and I'll split this one.

**Comment by SSE4** [comment]
> I could've created 3 separate issues so that PRs could have addressed those easier. However, I didn't want to spam the issue tracker with small cosmetic issues. If you still think creating 3 separate issues is preferable, let me know and I'll split this one.

that's no problem, feel free to create 3 separated issues.

**Comment by ghost** [comment]
Sure, I'll be right back with three separated issues :)

**Comment by memsharded** [comment]
> ERROR: Error while parsing [options] in conanfile
Options should be specified as 'pkg:option=value'

This has already been fixed for 2.0.1 in https://github.com/conan-io/conan/pull/13266


So we can use this single ticket for those 1, 2, they are simple enough, no need to waste more CI cycles than necessary, re-opening

**Comment by memsharded** [comment]
Closed by https://github.com/conan-io/conan/pull/13299 for 2.0.1
